desc/optimize/optimizer.py

```diff
@@ -148,8 +147,18 @@ def optimize(  # noqa: C901 - FIXME: simplify this
         """
         if not isinstance(constraints, (tuple, list)):
             constraints = (constraints,)
-        things = flatten_list(things, flatten_tuple=True)
+
+        # get unique things
+        things, indices = unique_list(flatten_list(things, flatten_tuple=True))
+        counts = np.unique(indices, return_counts=True)[1]
+        duplicate_idx = np.where(counts > 1)[0]
+        warnif(
+            len(duplicate_idx),
+            UserWarning,
+            f"{[things[idx] for idx in duplicate_idx]} is duplicated in things.",
```

Comment:
this is sort of esoteric, and do we want to throw an error if this is the case or does the rest of the optimizer work fine still if an object is duplicated in the passed-in things? We only use the unique things anyways?

Reply:
Everything should still work but it might lead to some weird behavior if things get updated twice etc